File: 754-reach-a-number/754-reach-a-number.cpp
class Solution {
public:
    int reachNumber(int target) {
        target = abs(target);
        long long n = ceil((-1 + sqrt(1 + 8.0*(target))) / 2);
        long long sum = (n * (n+1)) / 2;
        if(sum == target)   return n;
        long long rem = sum - target;
        if((rem&1) == 0)    return n;
        return n + ((n&1) ? 2 : 1);
    }
};

File: 754-reach-a-number/754-reach-a-number.py
class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        n = int(math.ceil((math.sqrt(8 * target + 1) - 1) / 2)) # solve inequation: n * (n + 1) / 2 >= target
        d = n * (1 + n) // 2 - target
        if d & 1 == 0:
            return n
        else:
            return n + (0 if n&1 == 0 else 1) + 1